docs/design/proposals/intree-migrate.md

```diff
@@ -68,8 +84,33 @@ field.
 field
 
 if `adminId` field is nil or not set, `UserID` field will be filled with default
-value ie `admin`.The above logic get activated only when the secret is a
+value ie `admin`. The above logic get activated only when the secret is a
 migration secret, otherwise skipped to the normal workflow as we have today.
+CreateVolume, DeleteVolume, NodeStage and ExpandVolume has to validate the
+secret and use for its operations:
+
+## Helper functions
+
+Few helper/utility functions will be introduced as part of this effort to handle
+migration specific logic to it.
+
+internal/rbd/migration.go
+
+* isMigrationVolID() - Identify the volume ID is of above-mentioned form of
+  migration volume ID.
+* parseMigrationVolID() - Parse the migration volume ID and populate the
+  required information like `pool`, `image`, `clusterID`..etc. into
+  `migrationVolID` structure.
+* deleteMigratedVolume() - Deletes the volume from the cluster by making use of
+  the `migrationVolID` structure fields like `pool, image, clusterID`.
+
+internal/util/credentials.go
+
+* IsMigrationSecret() - Identify the passed in secret is a migration secret
+  based on the map values (ex: `key` field) of request secret.
+* ParseAndSetSecretMapFromMigSecret() - Parse the secret map from the migration
```

Comment:
see https://github.com/ceph/ceph-csi/pull/2648/commits/70c6d268be82987d328c21bdeb0f1da52445d13d#r757268543 for a possible cleaner approach.

Reply:
@nixpanic added this change too.. ptal.. thanks :+1: